src/server/native_controllers/index.test.js

```diff
@@ -291,4 +291,19 @@ describe('#killOrStart', () => {
 
     await NativeControllers.killOrStart(kbnServer);
   });
+
+  test(`throws error when nativeController isn't for a known plugin`, () => {
+    const kbnServer = {
+      nativeControllers: [],
+      pluginSpecs: [{
+        getId: () => 'foo'
+      }],
+      disabledPluginSpecs: [],
+      nativeControllers: {
+        'bar': {}
+      }
+    };
+
+    return expect(NativeControllers.killOrStart(kbnServer)).rejects.toThrow();
```

Comment:
`toThrowMatchingSnapshot()`? Since it's just as easy :)

Reply:
This is news to me! I'll make it so.